src/qibo/hep/qpdf.py

```diff
@@ -0,0 +1,254 @@
+import numpy as np
+from qibo import models
+from qibo.hamiltonians import Hamiltonian, matrices
+from qibo import gates
+from qibo.config import raise_error
+
+
+class qPDF:
+    """Variational Circuit for Quantum PDFs (qPDF).
+
+    Args:
+        ansatz (str): the ansatz name, options are 'Weighted' and 'Fourier'.
+        layers (int): the number of layers for the ansatz.
+        nqubits (int): the number of qubits for the circuit.
+        multi_output (bool): allocates a multi-output model per PDF flavour (default is False).
+    """
+    def __init__(self, ansatz, layers, nqubits, multi_output=False):
+        """Initialize qPDF."""
+        if not isinstance(layers, int) or layers < 1: # pragma: no cover
+            raise_error(RuntimeError, "Layers must be positive and integer.")
+        if not isinstance(nqubits, int) or nqubits < 1: # pragma: no cover
+            raise_error(RuntimeError, "Number of qubits must be positive and integer.")
+        if not isinstance(multi_output, bool): # pragma: no cover
+            raise_error(TypeError, "multi-output must be a boolean.")
+
+        # load ansatz
+        try:
+            self.circuit, self.rotation, self.nparams = globals(
+            )[f"ansatz_{ansatz}"](layers, nqubits)
```

Comment:
I am not sure if there is a reason that the ansatz functions (`ansatz_Fourier` and `ansatz_Weighed`) are defined as globals (perhaps to be used outside the `qPDF` model?), however if they are moved within the class it would be possible to access them using `getattr(self, f"ansatz_{ansatz}")` which is usually safer.